desktop-exporter/server.go

```diff
@@ -13,51 +14,92 @@ type Server struct {
 	traceStore *TraceStore
 }
 
-func getTracesHandler(traceStore *TraceStore) func(http.ResponseWriter, *http.Request) {
+func newTraceSummary(traceID string, spans []SpanData) (*TraceSummary, error) {
+	durationMS, err := getTraceDuration(spans)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TraceSummary{
+		TraceID:    traceID,
+		SpanCount:  uint32(len(spans)),
+		DurationMS: durationMS,
+	}, nil
+}
+
+func newTraceData(store *TraceStore, traceID string) (*TraceData, error) {
+	spans, err := store.GetSpansByTraceID(traceID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TraceData{
+		Spans: spans,
+	}, nil
+}
+
+func getTracesHandler(store *TraceStore) func(http.ResponseWriter, *http.Request) {
```

Comment:
For the handlers, we could use `newTracesHandler`, or simplify it to just `tracesHandler`?

```suggestion
func tracesHandler(store *TraceStore) func(http.ResponseWriter, *http.Request) {
```